src/Sulu/Component/Content/Document/Subscriber/ResourceSegmentSubscriber.php

```diff
@@ -128,4 +151,16 @@ private function getResourceSegmentProperty($document)
 
         return $property;
     }
+
+    private function persistDocument(ResourceSegmentBehavior $document, PropertyMetadata $property)
```

Comment:
@dantleech Not sure about that... `Something that needs explaining` is a quite vague definition, it's always clear to the developer at the time of writing...